lib/ts/BaseLogFile.cc

```diff
@@ -0,0 +1,571 @@
+  : m_name(ats_strdup(name)), m_hostname(NULL), m_is_regfile(false), m_has_signature(false), m_signature(0), m_is_init(false)
+{
+  m_fp = NULL;
+  m_start_time = time(0);
+  m_end_time = 0L;
+  m_bytes_written = 0;
+  m_meta_info = NULL;
+
+  log_log_trace("exiting BaseLogFile constructor, m_name=%s, this=%p\n", m_name, this);
+}
+
+/*
+ * This consturctor creates a BaseLogFile based on a given name.
+ * Similar to above constructor, but is overloaded with the object signature
+ */
+BaseLogFile::BaseLogFile(const char *name, uint64_t sig)
+  : m_name(ats_strdup(name)), m_hostname(NULL), m_is_regfile(false), m_has_signature(true), m_signature(sig), m_is_init(false)
+{
+  m_fp = NULL;
+  m_start_time = time(0);
+  m_end_time = 0L;
+  m_bytes_written = 0;
+  m_meta_info = NULL;
+
+  log_log_trace("exiting BaseLogFile constructor, m_name=%s, this=%p\n", m_name, this);
+}
+
+/*
+ * This copy constructor creates a BaseLogFile based on a given copy.
+ */
+BaseLogFile::BaseLogFile(const BaseLogFile &copy)
+  : m_fp(NULL), m_start_time(copy.m_start_time), m_end_time(0L), m_bytes_written(0), m_name(ats_strdup(copy.m_name)),
+    m_hostname(ats_strdup(m_hostname)), m_is_regfile(false), m_has_signature(copy.m_has_signature), m_signature(copy.m_signature),
+    m_is_init(copy.m_is_init), m_meta_info(NULL)
+{
+  log_log_trace("exiting BaseLogFile copy constructor, m_name=%s, this=%p\n", m_name, this);
+}
+
+/*
+ * Destructor.
+ */
+BaseLogFile::~BaseLogFile()
+{
+  log_log_trace("entering BaseLogFile destructor, m_name=%s, this=%p\n", m_name, this);
+
+  if (m_is_regfile)
+    close_file();
+  else
+    log_log_trace("not a regular file, not closing, m_name=%s, this=%p\n", m_name, this);
+  if (m_name)
+    ats_free(m_name);
+  if (m_hostname)
+    ats_free(m_hostname);
+
+  log_log_trace("exiting BaseLogFile destructor, this=%p\n", this);
+}
+
+/*
+ * This function is called by a client of BaseLogFile to roll the underlying
+ * file  The tricky part to this routine is in coming up with the new file name,
+ * which contains the bounding timestamp interval for the entries
+ * within the file.
+
+ * Under normal operating conditions, this BaseLogFile object was in existence
+ * for all writes to the file.  In this case, the LogFile members m_start_time
+ * and m_end_time will have the starting and ending times for the actual
+ * entries written to the file.
+
+ * On restart situations, it is possible to re-open an existing BaseLogFile,
+ * which means that the m_start_time variable will be later than the actual
+ * entries recorded in the file.  In this case, we'll use the creation time
+ * of the file, which should be recorded in the meta-information located on
+ * disk.
+
+ * If we can't use the meta-file, either because it's not there or because
+ * it's not valid, then we'll use timestamp 0 (Jan 1, 1970) as the starting
+ * bound.
+
+ * Return 1 if file rolled, 0 otherwise
+ */
+int
+BaseLogFile::roll(long interval_start, long interval_end)
```

Comment:
It was originally like that in LogFile.cc. I assume for reasons related to the `snprintf()` calls later.